x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/MoveToErrorStepUpdateTask.java

```diff
@@ -73,25 +70,35 @@ public ClusterState execute(ClusterState currentState) {
     }
 
     @Override
-    public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
-        if (newState.equals(oldState) == false) {
-            stateChangeConsumer.accept(newState);
-        }
+    public void onClusterStateProcessed(ClusterState oldState, ClusterState newState) {
+        stateChangeConsumer.accept(newState);
     }
 
     @Override
-    public void onFailure(Exception e) {
-        final MessageSupplier messageSupplier = () -> new ParameterizedMessage(
-            "policy [{}] for index [{}] failed trying to move from step [{}] to the ERROR step.",
-            policy,
-            index.getName(),
-            currentStepKey
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        MoveToErrorStepUpdateTask that = (MoveToErrorStepUpdateTask) o;
+        // We don't have a stable equals on the cause and shouldn't have simultaneous moves to error step to begin with when deduplicating
+        // tasks so we only compare the current state here and in the hashcode.
+        return index.equals(that.index) && policy.equals(that.policy) && currentStepKey.equals(that.currentStepKey);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(index, policy, currentStepKey);
+    }
+
+    @Override
+    protected void handleFailure(Exception e) {
+        logger.warn(
+            new ParameterizedMessage(
+                "policy [{}] for index [{}] failed to to move from step [{}] to error step.",
+                policy,
+                index,
+                currentStepKey
+            ),
+            e
         );
-        if (ExceptionsHelper.unwrap(e, NotMasterException.class, FailedToCommitClusterStateException.class) != null) {
-            logger.debug(messageSupplier, e);
-        } else {
-            logger.error(messageSupplier, e);
-            assert false : new AssertionError("unexpected exception", e);
```

Comment:
I honestly don't remember or know why we have this assert, but are we okay dropping it?